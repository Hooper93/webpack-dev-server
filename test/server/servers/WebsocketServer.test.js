'use strict';

const http = require('http');
const express = require('express');
const ws = require('ws');
const WebsocketServer = require('../../../lib/servers/WebsocketServer');
const port = require('../../ports-map').WebsocketServer;

describe('WebsocketServer', () => {
  let socketServer;
  let listeningApp;

  beforeAll((done) => {
    // eslint-disable-next-line new-cap
    const app = new express();

    listeningApp = http.createServer(app);
    listeningApp.listen(port, 'localhost', () => {
      const server = {
        log: {
          error: () => {},
          debug: () => {},
        },
        sockPath: '/ws-server',
        listeningApp,
      };

      socketServer = new WebsocketServer(server);

      done();
    });
  });

  describe('server', () => {
    it('should recieve connection, send message, and close client', (done) => {
      const data = [];

      socketServer.onConnection((connection) => {
        data.push('open');
        socketServer.send(connection, 'hello world');
        setTimeout(() => {
          socketServer.close(connection);
        }, 1000);
      });

      // eslint-disable-next-line new-cap
      const client = new ws(`http://localhost:${port}/ws-server`);

      client.onmessage = (e) => {
        data.push(e.data);
      };

      client.onclose = () => {
        data.push('close');
      };

      setTimeout(() => {
        expect(data).toMatchSnapshot();
        done();
      }, 3000);
    });
  });

  afterAll((done) => {
    listeningApp.close(done);
  });
});
