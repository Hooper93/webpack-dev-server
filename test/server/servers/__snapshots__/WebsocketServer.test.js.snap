// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebsocketServer server should recieve connection, send message, and close client 1`] = `
Array [
  "open",
  "hello world",
  "close",
]
`;
